src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,277 @@
+		}
+
+		const propertyFilter = '-all-';
+
+		const options = {
+			method: 'GET',
+			// eslint-disable-next-line max-len
+			url: `${Configuration.get('ES_DOMAIN')
+			}/edu-sharing/rest/node/v1/nodes/mv-repo.schul-cloud.org/${id
+			}/metadata?propertyFilter=${propertyFilter}`,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		const eduResponse = await this.requestRepeater(options);
+		return eduResponse.node;
+	}
+
+	async FIND(data) {
+		const contentType = data.query.contentType || 'FILES'; // enum:[FILES,FILES_AND_FOLDERS,COLLECTIONS,ALL]
+		const skipCount = parseInt(data.query.$skip, 10) || 0;
+		const maxItems = parseInt(data.query.$limit, 10) || 9;
+		const sortProperties = data.query.sortProperties || 'score';
+		const sortAscending = false;
+		const propertyFilter = '-all-'; // '-all-' for all properties OR ccm-stuff
+		const searchWord = data.query.searchQuery || '';
+
+		if (!this.allConfigurationValuesHaveBeenDefined()) {
+			throw this.configurationIncompleteError();
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+
+		const urlBase = `${Configuration.get('ES_DOMAIN')
+		}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?`;
+		const url = urlBase
+			+ [
+				`contentType=${contentType}`,
+				`skipCount=${skipCount}`,
+				`maxItems=${maxItems}`,
+				`sortProperties=${sortProperties}`,
+				`sortAscending=${sortAscending}`,
+				`propertyFilter=${propertyFilter}`,
+			].join('&');
+
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			url,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] },
+				],
+				facettes: ['cclom:general_keyword'],
+			}),
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		const parsed = await this.requestRepeater(options);
+
+		// provided by client eg data.query.filterOptions
+		const filterOptions = {
+			mimetype: ['text/html', 'image/jpeg'],
+			provider: ['BauhausMaterial.de', 'München educationcenter', 'Khan Academy'],
+		};
+
+		// filter away everything buy selected mimetype
+		const filterMime = (obj, mimetypes) => {
+			let result = obj.nodes;
+			mimetypes.forEach((type) => { result = result.filter((n) => n.mimeType === type); });
+			return result;
+		};
+
+		// filter away everything buy selected providers
+		const filterProvider = (obj, providers) => obj === providers;
+
+		// adds accesstoken to image-url to let user see the picture on client-side.
+		if (parsed && parsed.nodes) {
+			parsed.nodes.forEach((node) => {
+				if (node.preview && node.preview.url) {
+					node.preview.url += `&accessToken=${this.accessToken}`;
+				}
+			});
+		}
+
+		const filterResult = (obj, opt) => {
```

Comment:
Codacy found an issue: ['filterResult' is assigned a value but never used.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4977295)